src/main/java/seedu/address/logic/commands/EditProjectCommand.java

```diff
@@ -0,0 +1,193 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRIORITY;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.project.Deadline;
+import seedu.address.model.project.Name;
+import seedu.address.model.project.Project;
+import seedu.address.model.project.ProjectPriority;
+
+/**
+ * Edits the details of an existing project in the TaskHub.
+ */
+public class EditProjectCommand extends Command {
+
+    public static final String COMMAND_WORD = "editP";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the project identified "
+            + "by the index number used in the displayed project list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NAME] "
+            + "[" + PREFIX_PRIORITY + "PRIORITY] "
+            + "[" + PREFIX_DEADLINE + "DEADLINE] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_PRIORITY + "HIGH "
+            + PREFIX_DEADLINE + "2020-12-31";
+
+    public static final String MESSAGE_EDIT_PROJECT_SUCCESS = "Edited Project: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_PROJECT = "This project already exists in the TaskHub.";
+
+    public final Index index;
+    public final EditProjectDescriptor editProjectDescriptor;
+
+    /**
+     * @param index of the project in the filtered project list to edit
+     * @param editProjectDescriptor details to edit the project with
+     */
```

Comment:
I think you may have forgotten the Javadoc description?

Reply:
There is nothing noteworthy to say here, and this is how it is written in `EditEmployeeCommand.java` as well.